openapi-validator/src/main/java/com/networknt/openapi/RequestValidator.java

```diff
@@ -123,7 +124,16 @@ private Status validatePathParameters(final NormalisedPath requestPath, final Op
             }
 
             final String paramName = openApiOperation.getPathString().paramName(i);
-            final String paramValue = requestPath.part(i);
+
+            String decodeParamValue = null;
+
+            try {
+                decodeParamValue = URLDecoder.decode(requestPath.part(i), "UTF-8");
+            } catch (Exception e) {
+                logger.info("Path parameter cannot be decoded, it will be used directly");
+            }
+
+            final String paramValue = (decodeParamValue == null) ? requestPath.part(i) : decodeParamValue;
```

Comment:
You might not have to create a new variable to store this intermediate value, perhaps something like this could work?:
```java
String paramValue = requestPath.part(i); // If it can't be UTF-8 decoded, use directly.
try {
    paramValue = URLDecoder.decode(requestPath.part(i), "UTF-8");
} catch (Exception e) {
    logger.info("Path parameter cannot be decoded, it will be used directly");
}
```

Reply:
Hi, I think what you said makes sense. My previous thought was to make this variable set to final and thus immutable. In case the value is changed during the validation process. Do you think it is necessary to be set to final?

Reply:
I dont think it would be necessary in this case.. since a string is immutable and not being overwritten in it's current context.. i dont think the variable being final or not would make a difference.